src/plans/nonlinear_least_squares_plan.jl

```diff
@@ -99,10 +99,13 @@ function get_gradient!(
 end
 
 function get_gradient!(
-    M::AbstractManifold, X, nlso::NonlinearLeastSquaresObjective{InplaceEvaluation}, p
+    M::AbstractManifold,
+    X,
+    nlso::NonlinearLeastSquaresObjective{InplaceEvaluation},
+    p,
+    Jval=zeros(nlso.num_components, manifold_dimension(M)),
```

Comment:
I am a bit confused by this new and even positional (not even keyword) argument.
It is not documented (yet), I am not 100% sure what it is for and would have more expected this to be an allocated field in `nlso` anyways – and an implementation detail of only this algorithm.

Is there a reason why it has to become part of the public interface of `get_gradient` where this positional argument is usually the index for the stochastic case and hence would cause ambiguities here?

Reply:
Yes, turning it into a keyword argument is a good idea.

Reply:
It's for storing Jacobian computed as a temporary.

Reply:
I do see the idea here and what you try to avoid, sure. I am mainly trying to introduce this properly into the structure of Manopt without introducing too much of a mix between a nice interface and the (user hidden) implementation technical foo.